src/steps/active-directory/converters.ts

```diff
@@ -0,0 +1,227 @@
+      },
+    },
+  });
+}
+
+export function createAccountEntityWithOrganization(
+  instance: IntegrationInstance,
+  organization: Organization,
+  intuneConfig: {
+    mobileDeviceManagementAuthority?: string;
+    subscriptionState?: string;
+    intuneAccountID?: string;
+  },
+): Entity {
+  let defaultDomain: string | undefined;
+  const verifiedDomains = organization.verifiedDomains?.map((e) => {
+    if (e.isDefault) {
+      defaultDomain = e.name as string | undefined;
+    }
+    return e.name as string;
+  });
+
+  return createIntegrationEntity({
+    entityData: {
+      source: {
+        organization,
+        intuneConfig,
+      },
+      assign: {
+        _class: entities.ACCOUNT._class,
+        _key: `${entities.ACCOUNT._type}-${instance.id}`,
+        _type: entities.ACCOUNT._type,
+        id: organization.id,
+        name: organization.displayName,
+        displayName: instance.name,
+        organizationName: organization.displayName,
+        defaultDomain,
+        verifiedDomains,
+        intuneAccountId: intuneConfig?.intuneAccountID,
+        mobileDeviceManagementAuthority:
+          intuneConfig?.mobileDeviceManagementAuthority,
+        intuneSubscriptionState: intuneConfig?.subscriptionState,
+      },
+    },
+  });
+}
+
+export function createMachineEntity(data: Machine): Entity {
+  return createIntegrationEntity({
+    entityData: {
+      source: {}, // removed due to size
+      assign: {
+        _class: entities.MACHINE._class,
+        _type: entities.MACHINE._type,
+        _key: data.id!,
+        id: data.id,
+        agentVersion: data.agentVersion,
+        defenderAvStatus: data.defenderAvStatus,
+        riskScore: data.riskScore,
+        name: data.computerDnsName,
+        computerDnsName: data.computerDnsName,
+        displayName: data.computerDnsName as string | undefined,
+        function: ['endpoint-protection', 'vulnerability-detection'],
+      },
+    },
+  });
+}
+
+export function createMachinesDeviceRelationship(data: Entity): Relationship {
+  return createMappedRelationship({
+    // source: data,
+    _class: RelationshipClass.MANAGES,
+    _type: MappedRelationships.MACHINE_MANAGES_DEVICE._type,
+    _mapping: {
+      sourceEntityKey: data._key,
+      relationshipDirection:
+        MappedRelationships.MACHINE_MANAGES_DEVICE.direction,
+      targetFilterKeys: [['_type', '_class', 'hostname']],
+      targetEntity: {
+        _class: TargetEntities.DEVICE._class,
+        _type: TargetEntities.DEVICE._type,
+        displayName: data.computerDnsName as string | undefined,
+        name: data.displayName,
+        id: data.id,
+        category: 'endpoint',
+        fqdn: data.computerDnsName as string | undefined,
+        hostname: data.computerDnsName as string | undefined,
+        osName: data.osPlatform as string | undefined,
+        osVersion: data.osVersion as string | undefined,
+        platform: data.osPlatform as string | undefined,
```

Comment:
I'm unsure what the intent is of `| undefined` when these are cast `as string` 🤔